src/ccover/ccover_parse_options.cpp

```diff
@@ -0,0 +1,753 @@
+    {
+      status() << "String Abstraction" << eom;
+      string_abstraction(
+        goto_model,
+        get_message_handler());
+    }
+
+    // add failed symbols
+    // needs to be done before pointer analysis
+    add_failed_symbols(goto_model.symbol_table);
+
+    // recalculate numbers, etc.
+    goto_model.goto_functions.update();
+
+    // add loop ids
+    goto_model.goto_functions.compute_loop_numbers();
+
+    if(cmdline.isset("drop-unused-functions"))
+    {
+      // Entry point will have been set before and function pointers removed
+      status() << "Removing unused functions" << eom;
+      remove_unused_functions(goto_model, get_message_handler());
+    }
+
+    // remove skips such that trivial GOTOs are deleted and not considered
+    // for coverage annotation:
+    remove_skip(goto_model);
+
+    // instrument cover goals
+    if(cmdline.isset("cover"))
+    {
+      if(instrument_cover_goals(options, goto_model, get_message_handler()))
+        return true;
+    }
+
+    // label the assertions
+    // This must be done after adding assertions and
+    // before using the argument of the "property" option.
+    // Do not re-label after using the property slicer because
+    // this would cause the property identifiers to change.
+    label_properties(goto_model);
+
+    // full slice?
+    if(cmdline.isset("full-slice"))
+    {
+      status() << "Performing a full slice" << eom;
+      if(cmdline.isset("property"))
+        property_slicer(goto_model, cmdline.get_values("property"));
+      else
+        full_slicer(goto_model);
+    }
+
+    // remove any skips introduced since coverage instrumentation
+    remove_skip(goto_model);
+  }
+
+  catch(const char *e)
+  {
+    error() << e << eom;
+    return true;
+  }
+
+  catch(const std::string &e)
+  {
+    error() << e << eom;
+    return true;
+  }
+
+  catch(int e)
+  {
+    error() << "Numeric exception : " << e << eom;
+    return true;
+  }
+
+  catch(const std::bad_alloc &)
+  {
+    error() << "Out of memory" << eom;
+    exit(CPROVER_EXIT_INTERNAL_OUT_OF_MEMORY);
+    return true;
+  }
+
+  return false;
+}
+
+/// display command line help
+void ccover_parse_optionst::help()
+{
+  // clang-format off
+  std::cout <<
+    "\n"
+    "* *  ccover " CBMC_VERSION " - Copyright (C) 2001-2018 ";
+
+  std::cout << "(" << (sizeof(void *)*8) << "-bit version)";
+
+  std::cout << "  * *\n";
+
+  std::cout <<
+    "* *                     Daniel Kroening                     * *\n"
+    "* *    University of Oxford Computer, Science Department    * *\n"
+    "* *                 kroening@kroening.com                   * *\n"
+    "\n"
+    "Usage:                       Purpose:\n"
+    "\n"
+    " ccover [-?] [-h] [--help]    show help\n"
+    " cccover file.c ...           source file names\n"
+    "\n"
+    "Test suite generation options:\n"
+    " --cover CC                   create test-suite with coverage criterion CC\n" // NOLINT(*)
```

Comment:
What is the default behaviour if this option is not given?

Reply:
now documented